app/controllers/developer_controller.rb

```diff
@@ -1,42 +1,72 @@
 class DeveloperController < ApplicationController
 
-  def show_developer
-
+  def show
     @developer = Developer.where(newuser_id: current_newuser.id)
-
-
-
   end
 
-  def create_developer
-
+  def create
+    @developer = Developer.new
   end
 
   def save_developer
-
-
-    @developer = Developer.new(specilization: params[:specilization], name: params[:name], age: params[:age], phoneNo: params[:phoneNo], email: params[:email], newuser_id: current_newuser.id )
+    # byebug
+    @developer = Developer.new(developer_params)
     if @developer.save()
       flash[:notice] = "Developer saved successfully"
       redirect_to user_developer_path
     else
       flash[:error] = @developer.errors.full_messages.to_sentence
       redirect_to user_developer_path
     end
+  end
 
+  def edit
+    @developer = find_developer()
 
   end
 
-  def update_developer
+  def update
+    @developer = find_developer()
+    if @developer.update(update_params)
+      flash[:notice] = "Developer updated successfully"
+      redirect_to user_developer_path
+    else
+      flash[:error] = @developer.errors.full_messages.to_sentence
+      redirect_to user_developer_path
+    end
+  end
+
+  def destory
+
+    @developer = find_developer()
+    if @developer.delete
+      flash[:success] = "Developer successfully deleted"
+      redirect_to user_developer_path
+    else
+      flash[:error] = @developer.errors.full_messages.to_sentencex
+      redirect_to user_developer_path
+    end
 
   end
 
-  def save_update_developer
 
+
+
+
+  private
+
+  def developer_params
+    d_params = params.require(:developer).permit(:specilization, :name, :age, :phoneNo, :email)
+    d_params[:newuser_id] = current_newuser.id
```

Comment:
this should be moved to the method in which developer_params are being used
This method should only be used to whitelist the incoming params and not to add new ones